runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -10570,6 +10570,340 @@ inlineCompareAndSwapNative(
    }
 
 
+/**
+ * \brief
+ *   Generate inlined instructions equivalent to java/lang/StringCoding.hasNegatives or java/lang/StringCoding.countPositives
+ *
+ * \param node
+ *   The tree node
+ *
+ * \param recognizedMethod
+ *   The method being inlined, should be either hasNegatives or countPositives
+ *
+ * \param cg
+ *   The Code Generator
+ */
+static TR::Register* inlineHasNegativesOrCountPositives(TR::Node* node, TR::RecognizedMethod recognizedMethod, TR::CodeGenerator* cg)
+   {
+   // Arguments to hasNegatives or countPositives
+   // Byte array
+   TR::Register *bufReg = cg->longClobberEvaluate(node->getChild(0));
```

Comment:
If I understand correctly, when a "clobber evaluate" is done, the code has to indicate explicitly when it's done using the register, so I think you'll a call to `cg->stopUsingRegister(bufReg)` at the end.

And again, if I understand correctly, you'll need to explicitly decrement the reference call for `node->getChild(0)` when you do a clobber evaluate on the node.

@0xdaryl, can you confirm that I have those requirements correct?

Reply:
> I do already decrement the reference count of all of node's children

Missed that.  Thanks!